cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java

```diff
@@ -1058,4 +1300,50 @@ public boolean apply(Privilege input) {
     Assert.assertTrue(Sets.filter(authorizer.listPrivileges(ALICE), entityFilter).isEmpty());
     Assert.assertTrue(Sets.filter(authorizer.listPrivileges(BOB), entityFilter).isEmpty());
   }
+
+  private void verifyDatasetIsEmpty(NamespaceId namespaceId, String datasetName) throws Exception {
+    DataSetManager<KeyValueTable> outTableManager = getDataset(namespaceId.toId(), datasetName);
+    KeyValueTable outputTable = outTableManager.get();
+    CloseableIterator<KeyValue<byte[], byte[]>> scanner = outputTable.scan(null, null);
+    Assert.assertFalse(scanner.hasNext());
+    scanner.close();
+  }
+
+  private void verifyProgramFailure(Map<String, String> programArgs, ProgramManager programManager) throws
+    ExecutionException, InterruptedException, TimeoutException {
+    programManager.start(programArgs);
+    programManager.waitForFinish(5, TimeUnit.MINUTES);
+
+    // verify program history just have failures
+    List<RunRecord> history = programManager.getHistory();
```

Comment:
Yes, but I want to make sure all runs are failed. If I do it that way I will have to unnecessary have to keep a count as how many failed records I expect.